R/deprecated.R

```diff
@@ -91,16 +91,16 @@ deprecate_soft <- function(when,
                            with = NULL,
                            details = NULL,
                            id = NULL,
-                           env = caller_env(2)) {
+                           env = caller_env()) {
```

Comment:
We probably should take both `env` and `user_env`, this way callers can wrap `deprecate_soft()` and still have it work.

Defaulting `env` to `caller_env()` instead of `caller_env(2)` seems right in theory in that context. However it is a breaking change because that argument was meant to be the user environment.

Reply:
Yeah, fortunately I don't think anyone has wrapped these functions.